### PR TITLE
Distinguish `.clamp()` and `.setValue()`

### PR DESCRIPTION
**Pull request by ms609**
Following on from #598 and a Slack discussion with @bredelings , I now have a better understanding of the distinction between `.clamp()` and `.setValue()`, which I've tried to encapsulate in documentation entries.  Perhaps the `clamp` example is a bit laboured, but I think it's useful to illustrate the distinction between the two methods.

### CODE REVIEW

**Comment by davidcerny** [comment]
It might be worth mentioning that `.setValue()` is often used to initialize the MCMC with plausible values. [This tutorial](https://revbayes.github.io/tutorials/divrate/branch_specific.html) has a nice example:

```
speciation_mean ~ dnLoguniform( 1E-6, 1E2)
extinction_mean ~ dnLoguniform( 1E-6, 1E2)
extinction_mean.setValue( speciation_mean / 2.0 )
```

**Comment by ms609** [comment]
I've rewritten the description to emphasize this, and added a link to the tutorial – it's a good idea to refer the user to a real-life example.

File: help/md/clamp.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# Set the observed value of x
+x.clamp(1)
+
+# Evaluate P(z) at x = X, y = 1
+y.setValue(1)
+z.probability()
+
+# Modify the observed value of x
+x.clamp(2) # equivalent to x.unclamp(); x.clamp(2)
+z.probability()
```
Comment (davidcerny) [inline help/md/clamp.md:32]: These lines don't work from me: on calling `z.probability()`, I get:

```
   Error:	Variable of type 'Real' has no method called 'probability'.  You can use '.methods()' to find available methods.
```

I suppose that makes sense – there is no mechanism for RevBayes to figure out what the PDF is for the product of two normally distributed random variables.

File: help/md/clamp.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+# Evaluate P(z) at a different value of y
+y.redraw()
+print(y)
+z.probability()
+
+# Because x is clamped, it is invalid to call x.redraw() or mvSlide(y)
+# x will remain constant during mcmc, whereas y will be inferred.
```
Comment (davidcerny) [inline help/md/clamp.md:40]: Very nitpicky, but perhaps capitalize "MCMC" here

**Review by davidcerny — CHANGES_REQUESTED** [review]

File: help/md/clamp.md (hunk 1 above)
Comment (ms609) [reply help/md/clamp.md]: Oof... thanks for catching that.  I've updated (and tested!) the example.

**Review by davidcerny — APPROVED** [review]

**Pull request merged**